shell/src/main/java/tachyon/shell/TFsShell.java

```diff
@@ -690,98 +627,97 @@ public int run(String[] argv) {
-        for (TachyonURI path : paths) {
-          try {
-            if (cmd.equals("cat")) {
-              exitCode |= cat(path);
-            } else if (cmd.equals("ls")) {
-              exitCode |= ls(path);
-            } else if (cmd.equals("lsr")) {
-              exitCode |= lsr(path);
-            } else if (cmd.equals("rm")) {
-              exitCode |= rm(path);
-            } else if (cmd.equals("rmr")) {
-              exitCode |= rmr(path);
-            } else if (cmd.equals("tail")) {
-              exitCode |= tail(path);
-            } else if (cmd.equals("load")) {
-              exitCode |= load(path);
-            } else if (cmd.equals("fileinfo")) {
-              exitCode |= fileinfo(path);
-            } else if (cmd.equals("location")) {
-              exitCode |= location(path);
-            } else if (cmd.equals("report")) {
-              exitCode |= report(path);
-            } else if (cmd.equals("pin")) {
-              exitCode |= pin(path);
-            } else if (cmd.equals("unpin")) {
-              exitCode |= unpin(path);
-            } else if (cmd.equals("free")) {
-              exitCode |= free(path);
-            } else if (cmd.equals("du")) {
-              exitCode |= du(path);
+          count(inputPath);
+        } else {
+          List<TachyonURI> paths = TFsShellUtils.getTachyonURIs(createFS(inputPath), inputPath);
+          if (paths.size() == 0) { // A unified sanity check on the paths
+            throw new IOException(inputPath + " does not exist.");
+          }
+          // We run the command on all the paths, continuing even if individual paths throw an
+          // exception. At the end, we throw the last exception that was raised, if any.
+          IOException ioe = null;
+          for (TachyonURI path : paths) {
+            try {
+              if (cmd.equals("cat")) {
+                cat(path);
+              } else if (cmd.equals("ls")) {
+                ls(path);
+              } else if (cmd.equals("lsr")) {
+                lsr(path);
+              } else if (cmd.equals("rm")) {
+                rm(path);
+              } else if (cmd.equals("rmr")) {
+                rmr(path);
+              } else if (cmd.equals("tail")) {
+                tail(path);
+              } else if (cmd.equals("load")) {
+                load(path);
+              } else if (cmd.equals("fileinfo")) {
+                fileinfo(path);
+              } else if (cmd.equals("location")) {
+                location(path);
+              } else if (cmd.equals("report")) {
+                report(path);
+              } else if (cmd.equals("pin")) {
+                pin(path);
+              } else if (cmd.equals("unpin")) {
+                unpin(path);
+              } else if (cmd.equals("free")) {
+                free(path);
+              } else if (cmd.equals("du")) {
+                du(path);
+              }
+            } catch (IOException e) {
+              ioe = e;
```

Comment:
The old behavior was to run the command on as many paths as possible. that seems to be what unix does, too, mostly. Printing the old error message sounds good, though.